pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,33 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// FindMatchingUntoleratedTaint checks if the given tolerations tolerates
+// all the filtered taints, and returns the first taint without a toleration
+func FindMatchingUntoleratedTaint(taints []v1.Taint, tolerations []v1.Toleration, applyFilter taintsFilterFunc) *v1.Taint {
```

Comment:
For the return value, please use `(v1.Taint, bool)` instead of the current `*v1.Taint`.

Why? A caller would reasonably expect to get a pointer to one of the items in the list, but this function actually returns a pointer to a copy of a taint which comes from a filtered list.

Reply:
in case there is no taint, would it then return an empty Taint struct along with the bool ?

Reply:
Right, exactly.

Reply:
fixed it